docs/DeveloperGuide.md

```diff
@@ -2,37 +2,186 @@
 
 ## Acknowledgements
 
-{list here sources of all reused/adapted ideas, code, documentation, and third-party libraries -- include links to the original source as well}
+Referred to https://se-education.org/addressbook-level3/DeveloperGuide.html when drafting the DG
 
 ## Design & implementation
 
-{Describe the design and implementation of the product. Use UML diagrams and short code snippets where applicable.}
+### Architecture
 
+![](./images/architeture.png)
+
+The Architecture Diagram given above explains the high-level design of the App.
+
+The **User** sends command which is processed by Input
+
+**Input**: Receives user input string, processes it and sends to CashLeh main app
+
+**CashLeh**: main app
+
+**UI**: Responsible for the printing onto CLI
+
+**Parser**: Process the input string and calls respective methods
+
+**FileStorage**: Reads data from, and writes data to file
+
+### How the architecture components interact with each other
+
+![](./images/main_example.png)
```

Comment:
Activation bar should not extend past the return arrow.
<img width="628" alt="image" src="https://github.com/nus-cs2113-AY2324S1/tp/assets/142559789/4181b875-c146-41ad-972b-aa7627d8b7d3">

Self invocation requires an activation bar as well.
<img width="745" alt="image" src="https://github.com/nus-cs2113-AY2324S1/tp/assets/142559789/e439bbd4-42ae-4a63-9e37-ed650e1cb861">